contracts/base/Dispatcher.sol

```diff
@@ -265,14 +269,26 @@ abstract contract Dispatcher is Payments, V2SwapRouter, V3SwapRouter, V3ToV4Migr
     /// @return output The resultant recipient for the command
     function map(address recipient) internal view returns (address) {
         if (recipient == Constants.MSG_SENDER) {
-            return msgSender();
+            return _msgSender();
         } else if (recipient == Constants.ADDRESS_THIS) {
             return address(this);
         } else {
             return recipient;
         }
     }
 
+    /// @notice Function to be used instead of msg.sender, as the contract performs self-reentrancy and at
+    /// times msg.sender == address(this). Instead _msgSender() returns the initiator of the lock
+    /// @dev overrides BaseActionsRouter._msgSender in V4Router
+    function _msgSender() internal view override returns (address) {
```

Comment:
can we just have 1 public function?

Reply:
i think yes but it would have to be named with an `_` to match BaseActionsRouter... unless youre happy to rename it in BAR to have no _

Reply:
happy to name in BAR to have no _ if its public

Reply:
ok we've made this change in periphery so when we merge in updated periphery this weekend i will change this!